x-pack/plugins/code/server/lsp/java_launcher.ts

```diff
@@ -94,28 +94,40 @@ export class JavaLauncher implements ILanguageServerLauncher {
     }
 
     let config = 'config_mac';
+    let bundledJavaHome = `${findJDK('osx')}/Contents/Home`;
     let javaPath = 'java';
     let javaHomePath = '';
     // detect platform
     switch (getOsPlatform()) {
       case 'darwin':
-        javaHomePath = `${findJDK('osx')}/Contents/Home`;
-        javaPath = path.resolve(`${javaHomePath}`, 'bin', 'java');
         break;
       case 'win32':
-        javaHomePath = `${findJDK('windows')}`;
-        javaPath = path.resolve(`${javaHomePath}`, 'bin', 'java.exe');
+        bundledJavaHome = `${findJDK('windows')}`;
         config = 'config_win';
         break;
       case 'linux':
-        javaHomePath = `${findJDK('linux')}`;
-        javaPath = path.resolve(`${javaHomePath}`, 'bin', 'java');
+        bundledJavaHome = `${findJDK('linux')}`;
         config = 'config_linux';
         break;
       default:
         log.error('Unable to find platform for this os');
     }
 
+    if (this.checkJavaRuntime()) {
+      javaHomePath = this.checkJavaRuntime();
+      if (!this.checkJavaVersion(javaHomePath)) {
+        javaHomePath = '';
+      }
+    }
+    if (!javaHomePath) {
```

Comment:
this works but the best practice for checking string emptiness in js is still

strValue === ""